marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ProgressListener.java

```diff
@@ -0,0 +1,202 @@
+package com.marklogic.client.datamovement;
+
+import com.marklogic.client.datamovement.QueryBatch;
+import com.marklogic.client.datamovement.QueryBatchListener;
+import com.marklogic.client.datamovement.QueryBatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+
+/**
```

Comment:
The class level javadoc needs more substance. At least an example of how to use it. If there are things the user should know, include that, too. For an example of what I'm talking about, take a look at DeleteListener or ExportListener.

Reply:
Got it, I'm working on ExportListener-like class comments now.